src/mumble/MainWindow.cpp

```diff
@@ -711,43 +693,38 @@ void MainWindow::hideEvent(QHideEvent *e) {
 		e->ignore();
 		return;
 	}
-#endif
-#ifndef Q_OS_MAC
-#	ifdef Q_OS_UNIX
-	if (!qApp->activeModalWidget() && !qApp->activePopupWidget())
-#	endif
-		if (Global::get().s.bHideInTray && qstiIcon->isSystemTrayAvailable() && e->spontaneous())
-			QMetaObject::invokeMethod(this, "hide", Qt::QueuedConnection);
 #endif
 	QMainWindow::hideEvent(e);
 }
 
 void MainWindow::showEvent(QShowEvent *e) {
-#ifndef Q_OS_MAC
-#	ifdef Q_OS_UNIX
-	if (!qApp->activeModalWidget() && !qApp->activePopupWidget())
-#	endif
-		if (Global::get().s.bHideInTray && qstiIcon->isSystemTrayAvailable() && e->spontaneous())
-			QMetaObject::invokeMethod(this, "show", Qt::QueuedConnection);
-#endif
 	QMainWindow::showEvent(e);
 }
 
 void MainWindow::changeEvent(QEvent *e) {
-	QWidget::changeEvent(e);
-
-#ifdef Q_OS_MAC
-	// On modern macOS/Qt combinations, the code below causes Mumble's
-	// MainWindow to not be interactive after returning from being minimized.
-	// (See issue mumble-voip/mumble#2171)
-	// So, let's not do it on macOS.
+	// Hide in tray when minimized
+	if (Global::get().s.bHideInTray && e->type() == QEvent::WindowStateChange) {
+		// This code block is not triggered on (X)Wayland due to a Qt bug we can do nothing about (QTBUG-74310)
+		QWindowStateChangeEvent *windowStateEvent = static_cast< QWindowStateChangeEvent * >(e);
+		if (windowStateEvent) {
+			bool wasMinimizedState = (windowStateEvent->oldState() & Qt::WindowMinimized) != 0;
+			bool isMinimizedState  = (windowState() & Qt::WindowMinimized) != 0;
```

Comment:
Technically, the `!= 0` check is not needed - also the flags could be marked as `const`